service/src/main/java/uk/nhs/adaptors/gp2gp/tasks/TaskExecutorFactory.java

```diff
@@ -0,0 +1,20 @@
+package uk.nhs.adaptors.gp2gp.tasks;
+
+import java.util.Map;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import lombok.RequiredArgsConstructor;
+
+@Component
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class TaskExecutorFactory {
+    private static final Map<String, TaskExecutor> TASK_EXECUTOR_MAP = Map.of(GetGpcDocumentTaskDefinition.class.toString(),
```

Comment:
This map needs to be updated every time we add a task. It is possible to do this using dependency injection instead.

Reply:
This is changed in https://github.com/nhsconnect/integration-adaptor-gp2gp/pull/27